### DISCUSSION TITLE
Can we use CDS and Clr components side by side?

### DISCUSSION THREAD

**Discussion by kumar-tadepalli** [vmware-clarity/ng-clarity#423] [Q&A] [answered]
We are thinking to use clr/angular as primary and use cds components(which were missing from clr).
Can we use CDS and Clr components side by side?
Do we see any css/other conflicts?

**Comment by bbogdanov** [comment, marked as answer]
Hi, you can.

That's the main idea.

There are stuff you need to make sure you definitely use in your code like the 'clr-ui.shim.css'

I would also advice using '@cds/angular' since it automates some manual steps for you while using the web components.

Other than that it should be fine. 

The only annoying but really helpful thing will be to manage the state since clarity core components are stateless and you control the state for each component you are using. E.g an input is invalid and you have to make sure that you are checking when that happens and to show an error message

**Reply by kumar-tadepalli** [reply]
That's awesome!

Can you elaborate on 'clr-ui.shim.css' and usage?
Could you share a reference file/project example?

**Comment by pdaigle** [comment]
Basic instructions for using the Core CSS shim with clarity-ng are available here: https://storybook.core.clarity.design/?path=/story/themes-clarity-ui--page
